src/components/structures/grouper/LateEventGrouper.ts

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixEvent } from "matrix-js-sdk/src/matrix";
+
+const UNSIGNED_KEY = "io.element.late_event";
+
+interface UnsignedLateEventInfo {
+    received_at: number;
+    group_id: string;
+}
+
+export function getLateEventInfo(mxEvent: MatrixEvent): UnsignedLateEventInfo | undefined {
```

Comment:
JSDoc comment please. a "late event" is not a concept that is widely understood I believe.